src/main/java/seedu/address/logic/commands/EditClientCommand.java

```diff
@@ -109,13 +108,18 @@ public CommandResult execute(Model model) throws CommandException {
         if (personToEdit.getCar() != null) {
             Car carToEdit = personToEdit.getCar();
             editedCar = createEditedCar(carToEdit, editCarDescriptor);
-            if (carToEdit.equals(editedCar) && model.hasCar(editedCar) && isCarEdited) {
+            if ((carToEdit.equals(editedCar) || model.hasCar(editedCar)) && isCarEdited) {
                 throw new CommandException(MESSAGE_DUPLICATE_CAR);
             }
         }
 
         Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor, editedCar);
-        if (personToEdit.equals(editedPerson) && model.hasPerson(editedPerson) && isPersonEdited) {
+
+        if (personToEdit.equals(editedPerson) && isPersonEdited) {
+            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
```

Comment:
I think this should throw a different error message instead of duplicate person, something that tells the user that the current user is the same as the one they are trying to edit to